services/sidecar/src/simcore_service_sidecar/core.py

```diff
@@ -21,12 +21,31 @@
 from .executor import Executor
 from .rabbitmq import RabbitMQ
 from .utils import execution_graph, find_entry_point, is_node_ready
+from .db import DBContextManager
 
 log = get_task_logger(__name__)
 log.setLevel(config.SIDECAR_LOGLEVEL)
 node_port_log.setLevel(config.SIDECAR_LOGLEVEL)
 
 
+async def does_task_require_gpu(node_id: str) -> bool:
+    """Checks if the comp_task's image field if it requires to use the GPU"""
+    async with DBContextManager() as db_engine:
```

Comment:
The engine in the db should not be started and close everytime. There should rather be an engine per application and connections acquired per request

Reply:
I am unsure on how this would mix with the Celery part. Previously the DBContextManager is acquired once per Celery job. Even though this context manager is embedded in the `does_task_require_gpu ` function, it is only called once per Celery job.